miio/airhumidifier_jsq.py

```diff
@@ -0,0 +1,330 @@
+import enum
+import logging
+from typing import Any, Dict
+
+import click
+
+from .airhumidifier import AirHumidifierException
+from .click_common import EnumType, command, format_output
+from .device import Device, DeviceInfo, Optional
+
+_LOGGER = logging.getLogger(__name__)
+
+# Xiaomi Zero Fog Humidifier
+MODEL_HUMIDIFIER_JSQ001 = "shuii.humidifier.jsq001"
+
+# Array of properties in same order as in humidifier response
+AVAILABLE_PROPERTIES = {
+    MODEL_HUMIDIFIER_JSQ001: [
+        "temperature",  # (degrees, int)
+        "humidity",  # (percentage, int)
+        "mode",  # ( 0: Intelligent, 1: Level1, ..., 5:Level4)
+        "buzzer",  # (0: off, 1: on)
+        "child_lock",  # (0: off, 1: on)
+        "led_brightness",  # (0: off, 1: low, 2: high)
+        "power",  # (0: off, 1: on)
+        "no_water",  # (0: enough, 1: add water)
+        "lid_opened",  # (0: ok, 1: lid is opened)
+    ]
+}
+
+
+class OperationMode(enum.Enum):
+    Intelligent = 0
+    Level1 = 1
+    Level2 = 2
+    Level3 = 3
+    Level4 = 4
+
+
+class LedBrightness(enum.Enum):
+    Off = 0
+    Low = 1
+    High = 2
+
+
+class AirHumidifierStatus:
+    """Container for status reports from the air humidifier jsq."""
+
+    def __init__(self, data: Dict[str, Any], device_info: DeviceInfo) -> None:
+        """
+        Status of an Air Humidifier (shuii.humidifier.jsq001):
+            [24, 30, 1, 1, 0, 2, 0, 0, 0]
+
+        Parsed by AirHumidifierJsq device as:
+            {'temperature': 24, 'humidity': 29, 'mode': 1, 'buzzer': 1,
+             'child_lock': 0, 'led_brightness': 2, 'power': 0, 'no_water': 0,
+              'lid_opened': 0}
+        """
+        self.data = data
+        self.device_info = device_info
+
+    @property
+    def power(self) -> str:
+        """Power state."""
+        return "on" if self.data["power"] == 1 else "off"
+
+    @property
+    def is_on(self) -> bool:
+        """True if device is turned on."""
+        return self.power == "on"
```

Comment:
Is there a need for `power` at all? This could be `bool(self.data["power"])` otherwise.

Reply:
To be honest I don't have any idea do we need a separate `power` method or not. I tried to make API similar to other humidifiers in implementation, they use the same pattern with `power` method. So should I get rid of it?

Reply:
We can leave it as it is, maybe the API will cleaned up at some point. I think personally that it's not really needed, as `is_on` boolean is much nicer API-wise.